src/components/CreateNimi/partials/NimiLinkField/NimiLinkField.tsx

```diff
@@ -2,21 +2,55 @@ import { Nimi, NimiLink, NimiLinkBaseDetails } from 'nimi-card';
 import { useFormContext } from 'react-hook-form';
 import { ChangeEventHandler, useEffect, useState } from 'react';
 
-import { Input, Label } from '../../../form';
+import { Label } from '../../../form';
+import { isValidUrl } from '../../../../utils';
+import styled from 'styled-components';
 
 export interface NimiLinkFieldProps {
   link: NimiLink;
   label: string;
 }
+const InputWrap = styled.div<{ isInputFocused: boolean; isError: boolean }>`
+  display: flex;
+  border: 2px solid ${({ isInputFocused, isError }) => (isError ? 'red' : isInputFocused ? 'blue' : '#e6e8ec')};
+  border-radius: 12px;
+`;
+const AtFields = styled.div`
+  background: #e6e8ec;
+  border-radius: 10px 0 0 10px;
+  padding: 12px 16px;
+  color: #777e91;
+`;
+const StyledInput = styled.input`
+  padding: 12px 16px;
+  flex: 1;
+  border: none;
+  background-color: transparent;
+  resize: none;
+  outline: none;
+  ::placeholder,
+  ::-webkit-input-placeholder {
+    color: #b1b5c3;
+  }
+`;
+const ErroText = styled.div`
+  color: red;
+  margin-top: 5px;
+`;
 
 /**
  * Handles the input for the link type
  */
 export function NimiLinkField({ link, label }: NimiLinkFieldProps) {
   const { setValue, getValues } = useFormContext<Nimi>();
   const [inputValue, setInputValue] = useState('');
+  const [isMyInputFocused, setIsMyInputFocused] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const hasAtField = link === 'twitter' || link === 'instagram' || link === 'telegram';
 
   const onChange: ChangeEventHandler<HTMLInputElement> = (event) => {
+    if (isValidUrl(event.target.value) && link !== 'website') setIsError(true);
+    else setIsError(false);
```

Comment:
I don't follow this rule, elaborate?

Reply:
so checks if typed value is url except for website since it should be an url. If it is url it will display error text and give red outline....I can handle it multiple ways but think this is most elegant